### PR TITLE
Set sane timeouts and a friendly error message when the Mesos leader is unavailable

### PR DESCRIPTION
**Pull request by solarkennedy**
The big win here is adding a timeout here. 
When running from a devbox it "only" fails in 15 seconds (3 * 5) instead of 15 minutes.

### CODE REVIEW

File: paasta_tools/cli/cmds/spark_run.py [hunk 1]
```diff
@@ -423,8 +424,15 @@ def get_spark_config(
         user_args["spark.eventLog.enabled"] = "true"
         user_args["spark.eventLog.dir"] = default_event_log_dir
 
+    try:
+        mesos_address = find_mesos_leader(args.cluster)
+    except MesosLeaderUnavailable:
+        print(
+            "Couldn't reach the Mesos leader from here. Are you running this command from the right environment?",
```
Comment (kawaiwanyelp) [inline paasta_tools/cli/cmds/spark_run.py]: nit: "right environment" may not be obvious enough for developers. I'd put something like "(e.g. attempting to access prod from dev)".

File: paasta_tools/mesos_tools.py [hunk 2]
```diff
@@ -165,7 +170,12 @@ def find_mesos_leader(cluster):
         raise ValueError("Mesos master is required to find leader")
 
     url = f"http://{master}:{MESOS_MASTER_PORT}/redirect"
-    response = requests.get(url)
+    try:
+        # Timeouts here are for connect, read
+        response = requests.get(url, timeout=(5, 30))
+    except Exception as e:
+        print(f"Error finding the mesos leader: {e}", file=sys.stderr)
+        raise MesosLeaderUnavailable(e)
```
Comment (kawaiwanyelp) [inline paasta_tools/mesos_tools.py]: Embedding the original exception inside `MesosLeaderUnavailable` will obscure the original exception in a print-out. I think it should be `raise MesosLeaderUnavailable() from e` instead.

Comment (solarkennedy) [reply paasta_tools/mesos_tools.py]: I don't really understand. My motivation here is that the original error (timeout, failed to connect, whatever it is) still gets printed to stderr in some way.

If I do `raise MLU from e` here, would you give a shipit if I printed the `e` in paasta spark run?

Comment (kawaiwanyelp) [reply paasta_tools/mesos_tools.py]: My motivation for chaining here is that I'd like to see the original exception class itself printed as well as well as the traceback to see where it happened. Often times, logging only the message is not quite enough for debugging, though this case it's less important.

Comment (kawaiwanyelp) [reply paasta_tools/mesos_tools.py]: Hm, given the output, I'm fine without the chaining. As long as the original exception is visible, even without a traceback, I'm happy with it.

**Pull request merged**